src/components/search-by-person-attributes/search-by-person-attributes.component.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useState } from "react";
+
+import { showNotification } from "@openmrs/esm-framework";
+import { Column, Dropdown, TextInput } from "carbon-components-react";
+import { useTranslation } from "react-i18next";
+
+import { PersonAttribute, SearchByProps } from "../../types";
+import SearchButtonSet from "../search-button-set/search-button-set";
+import { getPersonAttributes } from "./search-by-person-attributes.resource";
+import styles from "./search-by-person-attributes.style.scss";
+import {
+  getQueryDetails,
+  getSearchByAttributesDescription,
+} from "./search-by-person-attributes.utils";
+
+const SearchByPersonAttributes: React.FC<SearchByProps> = ({
+  onSubmit,
+  isLoading,
+}) => {
+  const { t } = useTranslation();
+  const [personAttributes, setPersonAttributes] = useState<PersonAttribute[]>(
+    []
+  );
+  const [selectedAttributeValues, setSelectedAttributeValues] = useState([]);
+  const [selectedAttributeId, setSelectedAttributeId] = useState<string>(null);
+
+  const onPersonAttributesReceive = async () => {
+    try {
+      setPersonAttributes(await getPersonAttributes());
+    } catch (error) {
+      showNotification({
+        title: t("error", "Error"),
+        kind: "error",
+        critical: true,
+        description: error?.message,
+      });
+    }
+  };
+
+  useEffect(() => {
+    onPersonAttributesReceive();
+  }, []);
```

Comment:
The warning is still there. What are your findings?

Reply:
The main purpose of the exhaustive-deps warning is to prevent the developers from missing dependencies inside their effect and losing some behaviour.

In this scenario, this is the behaviour I want so I guess we can ignore the warning here